### ISSUE TITLE
Superfluous whitespace

### ISSUE THREAD

**Issue by llaville** [squizlabs/PHP_CodeSniffer#317]
Hello Greg,

I use the latest version 2.0.0RC4 of phpcs, and I noticed something I think will be nice to catch to enhance the superfluous whitespace sniff.

If you've code with spaces line content, it's not detected as a violation (neither an error or a warning).

Here is an example : 

``` php
<?php
if (is_int($var)) {
    return $var; 
}

if (empty($var['analysers'])) {
}
```

At end of `return $var` there are extra whitespace, and also some white spaces on blank line between each if statements.

So if I run the following command 

```
phpcs -vvv --standard=PSR2  /path/to/sample/script
```

 I got 

```
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 3 | ERROR | Whitespace found at end of line
--------------------------------------------------------------------------------
```

I expect to have

```
--------------------------------------------------------------------------------
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 3 | ERROR | Whitespace found at end of line
 5 | ERROR | Whitespace found at end of line
--------------------------------------------------------------------------------
```

See token 17 about line 3
See token 20 corresponding to line 5 with extra spaces

Here is a chunk of console output

```
   *** START LEVEL MAP ***
    Process token 0 on line 1 [lvl:0;]: T_OPEN_TAG => <?php\n
    Process token 1 on line 2 [lvl:0;]: T_IF => if
    Process token 2 on line 2 [lvl:0;]: T_WHITESPACE =>
    Process token 3 on line 2 [lvl:0;]: T_OPEN_PARENTHESIS => (
    Process token 4 on line 2 [lvl:0;]: T_STRING => is_int
    Process token 5 on line 2 [lvl:0;]: T_OPEN_PARENTHESIS => (
    Process token 6 on line 2 [lvl:0;]: T_VARIABLE => $var
    Process token 7 on line 2 [lvl:0;]: T_CLOSE_PARENTHESIS => )
    Process token 8 on line 2 [lvl:0;]: T_CLOSE_PARENTHESIS => )
    Process token 9 on line 2 [lvl:0;]: T_WHITESPACE =>
    Process token 10 on line 2 [lvl:0;]: T_OPEN_CURLY_BRACKET => {
    => Found scope opener for 1:T_IF
            * level increased *
            * token 1:T_IF added to conditions array *
            Process token 11 on line 2 [lvl:1;conds;T_IF;]: T_WHITESPACE => \n
            Process token 12 on line 3 [lvl:1;conds;T_IF;]: T_WHITESPACE =>
            Process token 13 on line 3 [lvl:1;conds;T_IF;]: T_RETURN => return
            Process token 14 on line 3 [lvl:1;conds;T_IF;]: T_WHITESPACE =>
            Process token 15 on line 3 [lvl:1;conds;T_IF;]: T_VARIABLE => $var
            Process token 16 on line 3 [lvl:1;conds;T_IF;]: T_SEMICOLON => ;
            Process token 17 on line 3 [lvl:1;conds;T_IF;]: T_WHITESPACE =>  \n
            Process token 18 on line 4 [lvl:1;conds;T_IF;]: T_CLOSE_CURLY_BRACKET => }
            => Found scope closer for 10:T_OPEN_CURLY_BRACKET
            * token T_IF removed from conditions array *
            * level decreased *
    Process token 19 on line 4 [lvl:0;]: T_WHITESPACE => \n
    Process token 20 on line 5 [lvl:0;]: T_WHITESPACE =>     \n
    Process token 21 on line 6 [lvl:0;]: T_IF => if
```

Hope my explains are enough !
Laurent
